File: web/src/api/cronTask.js
import service from '@/utils/request'

// @Router /cron-task/config/list [post]
export const configList = (data) => {
    return service({
        url: '/cron-task/config/list',
        method: 'post',
        data
    })
}

// @Router /cron-task/config/add [post]
export const configAdd = (data) => {
    return service({
        url: '/cron-task/config/add',
        method: 'post',
        data
    })
}

// @Router /cron-task/config/modify [post]
export const configModify = (data) => {
    return service({
        url: '/cron-task/config/modify',
        method: 'post',
        data
    })
}

// @Router /cron-task/log/list [post]
export const logList = (data) => {
    return service({
        url: '/cron-task/log/list',
        method: 'post',
        data
    })
}

File: server/router/cron_task/cron_task.go
package cron_task

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type CronTaskRouter struct {
}

func (s *CronTaskRouter) InitApiRouter(Router *gin.RouterGroup) {
	apiRouter := Router.Group("cron-task")
	apiRouter.Use(middleware.OperationRecord())
	{
		configGroup := apiRouter.Group("config")
		{
			configGroup.POST("list", cronTaskApi.ConfigList)
			configGroup.POST("add", cronTaskApi.ConfigAdd)
			configGroup.POST("modify", cronTaskApi.ConfigModify)
		}

		logGroup := apiRouter.Group("log")
		{
			logGroup.POST("list", cronTaskApi.LogList)
		}
	}
}
